scaleway/resource_k8s_cluster.go

```diff
@@ -222,6 +222,34 @@ func resourceScalewayK8SCluster() *schema.Resource {
 				Description: "The status of the cluster",
 			},
 		},
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+			autoUpgradeEnable, okAutoUpgradeEnable := diff.GetOkExists("auto_upgrade.0.enable")
+			_, okAutoUpgradeStartHour := diff.GetOkExists("auto_upgrade.0.maintenance_window_start_hour")
+			_, okAutoUpgradeDay := diff.GetOk("auto_upgrade.0.maintenance_window_day")
+
+			if okAutoUpgradeEnable {
+				// check if either all or none of the auto upgrade attribute are set.
+				// if one auto upgrade attribute is set, they all must be set.
+				// if none is set, auto upgrade attributes will be computed.
+				if !(okAutoUpgradeDay && okAutoUpgradeStartHour) {
+					return fmt.Errorf("all field or zero field of auto_upgrade must be set")
+				}
+			}
+
+			version := diff.Get("version").(string)
+			versionIsOnlyMinor := len(strings.Split(version, ".")) == 2
+
+			clusterAutoUpgradeEnabled := false
+
+			if okAutoUpgradeDay && okAutoUpgradeEnable && okAutoUpgradeStartHour {
+				clusterAutoUpgradeEnabled = autoUpgradeEnable.(bool)
+				if versionIsOnlyMinor != clusterAutoUpgradeEnabled {
```

Comment:
why not just `if versionIsOnlyMinor != autoUpgradeEnable.(bool)` ?

Reply:
ok, I've just removed it because it is already. checked on the schema